packages/cluster-ui/src/transactionsPage/utils.ts

```diff
@@ -124,12 +125,52 @@ const withFingerprint = function(
   };
 };
 
+function addExecStats(a: ExecStats, b: ExecStats): Required<ExecStats> {
+  let execStatCountA = FixLong(a.count).toInt();
+  const execStatCountB = FixLong(b.count).toInt();
+  if (execStatCountA === 0 && execStatCountB === 0) {
+    // If both counts are zero, artificially set the one count to one to avoid
+    // division by zero when calculating the mean in addNumericStats.
+    execStatCountA = 1;
+  }
+  return {
+    count: a.count.add(b.count),
+    network_bytes:
+      a.network_bytes && b.network_bytes
+        ? aggregateNumericStats(
+            a.network_bytes,
+            b.network_bytes,
+            execStatCountA,
+            execStatCountB,
+          )
+        : null,
+    max_mem_usage:
+      a.max_mem_usage && b.max_mem_usage
+        ? aggregateNumericStats(
+            a.max_mem_usage,
+            b.max_mem_usage,
+            execStatCountA,
+            execStatCountB,
+          )
+        : null,
+    contention_time:
+      a.contention_time && b.contention_time
+        ? aggregateNumericStats(
+            a.contention_time,
+            b.contention_time,
+            execStatCountA,
+            execStatCountB,
+          )
+        : null,
+  };
+}
+
 // addTransactionStats adds together two stat objects into one using their counts to compute a new
 // average for the numeric statistics. It's modeled after the similar `addStatementStats` function
 function addTransactionStats(
   a: TransactionStats,
   b: TransactionStats,
-): TransactionStats {
+): Required<TransactionStats> {
```

Comment:
I added this in because I thought that incorrect addition of transaction stats was causing some null property accesses, but it was in fact due to something else. However, I kept it in because I thought it's hygienic. When is this used? I didn't update my statements page PR to update the similar `addStatementStats` (however, something similar exists in the main cockroach repo, which is not the case for transactions). What's the consequence of this? Should I update `addStatementStats`?

Reply:
Yes I like making changes like this. I think making the same change to `addStatementStats` is worth it.

Because the Statements Page has been integrated into CC, it has its own redux layer in `cluster-ui` that was built to work nicely with CC's redux store. That redux layer is *not* used with DB Console so there is some code duplication there. I think what we'll end up doing in the near term is moving the redux layer directly into CC to clarify that the two redux configurations are separate and make peace with some code duplication.

So making that change to `addStatementStats` will protect CC's version of the Statements Page from null lookups.